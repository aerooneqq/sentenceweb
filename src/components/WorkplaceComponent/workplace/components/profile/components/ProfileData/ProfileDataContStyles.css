
#profileDataOutterContainer{ 
  height: 90%;
  margin-right: 20px;
}

#profileDataContainer{
  width: 300px;
  height: 100%;

  background:  transparent;
  display: flex;
  flex-direction: column;

  background: #f9f9f9;
  border-style: none;
  border-radius: 10px;
}

#profileDataHeaderCont{
  display: flex;
}

#profileDataContentCont{ 
  height: 90%;
  width: 100%;

  background-color: transparent;

  display: flex;
  flex-grow: 1;
  flex-direction: column;

  margin-top: 20px;
}

.fadeInAnimation{ 
  animation: fadeIn 0.5s linear;
}

@keyframes fadeIn{ 
  from { 
    opacity: 0;
  }

  to { 
    opacity: 1;
  }
}

#profileDataContentCont .textBlock{ 
  margin-bottom: 15px;
}

#saveOrDiscardBlock{ 
  margin-bottom: 40px;
}

#discardCont{ 
  margin-top: 10px;
}

#profileDataLoaderContainer{ 
  width: 100%;
  height: 100%;

  display: flex;

  flex-direction: column;
  justify-content: center;
  align-items: center; 
}