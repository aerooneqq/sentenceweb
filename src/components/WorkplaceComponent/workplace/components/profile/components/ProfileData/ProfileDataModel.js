import React from "react";

//Profile data components.
import CareerData from "./profileDataAtoms/CareerData/CareerData";
import AuthProfileData from "./profileDataAtoms/AuthProfileData/AuthProfileData";
import NameData from "./profileDataAtoms/NameData/NameData";
import LocationData from "./profileDataAtoms/LocationData/LocationData";
import Friends from "./profileDataAtoms/subscribersAndSubscriptions/Friends/Friends";

/**
 * This class represents the model of a profile data. Profile data is devided in 5 sections: career data, authentication data,
 * name data, location data and frinds data (subscribers and subscriptions).
 */
export default class ProfileDataModel { 

    setUpdateUserFunction(func) { 
        this.updateUserFunction = func;

        return this;
    }

    configureModel() { 
        this.components = [ 
            {name: "Career data", component: <CareerData user = {this.user} changeUpdatedUser = {this.updateUserFunction} />},
            {name: "Authentication data", component:<AuthProfileData user = {this.user} changeUpdatedUser = {this.updateUserFunction} />},
            {name: "Name data", component:<NameData user = {this.user} changeUpdatedUser = {this.updateUserFunction} />},
            {name: "Location data", component:<LocationData user = {this.user} changeUpdatedUser = {this.updateUserFunction} />},
            {name: "Subscrivers and subscriptions", component:<Friends user = {this.user} changeUpdatedUser = {this.updateUserFunction} />},
        ];
    }

    /**
     * Updates the given property of the user.
     * @param {The name of the property which was updated} propertyName 
     * @param {The new value of the updated property} value 
     */
    changeUpdatedUser(propertyName, value) {
        this.user[propertyName] = value;
    }

    /**
     * Discards all changes in the _user object.
     */
    discardAllChanges() { 
        for (let key in this._user) { 
            this.user[key] = this.initialUser[key];
        }
    }

    /**
     * Returns the component which is under the given index.
     * @param {The index of the profile data} index 
     */
    getProfileDataContent(index) { 
        if (index < 0 || index >= this.components.length) { 
            return null;
        }

        return this.components[index].component;
    }

    /**
     * Gets the name of the profile data component under the given index.
     * @param {The index of the profile data} index 
     */
    getProfileDataName(index) { 
        if (index < 0 || index >= this.components.length) { 
            return null;
        }
        
        return this.components[index].name;
    }
}