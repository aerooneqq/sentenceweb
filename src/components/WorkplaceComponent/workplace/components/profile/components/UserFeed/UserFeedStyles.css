@media screen and (min-width: 400px){ 
    #userFeedContainer{ 
        height: 450px;
        width: 400px;
        background: transparent;
        margin-bottom: 30px;
    }
    
    #userFeedInnerCont{ 
        height: 410px;
        width: 100%;
    
        background:#f9f9f9;
    
        border-style: none; 
        border-radius: 10px;
    }
    
    #userFeedInnerScroll{ 
        width: 100%;
        height: 360px;
    
        overflow-y: auto;
    
        background:#f9f9f9;
        border-style: none; 
        border-top-left-radius: 10px; 
        border-top-right-radius: 10px;
    }
    
    #userFeedInnerScroll::-webkit-scrollbar{ 
        width: 3px;
    }
    
    #userFeedInnerScroll::-webkit-scrollbar-thumb{
        background:rgba(255,110,74, 0.5) !important; 
        border-radius: 2px; 
    }
    
    #userFeedInnerScroll::-webkit-scrollbar{ 
        height: 0px;
    }   
    
    #userFeedInputCont{ 
        width: 100%;
        height: 53px;
    
        background:white;
    
        border-style: none ; 
        border-width: 1px;
        border-color: #F97646;
        border-bottom-left-radius: 10px;
        border-bottom-right-radius: 10px;
    
        display: flex; 
        flex-direction: row; 
        justify-content: left;
        align-items: center;
    }
    
    #userFeedTextArea{ 
        width:300px;
        height: 53px;
    
        border-style: solid; 
        border-bottom-left-radius: 10px;
        border-width: 1px; 
        border-color: #F97646;
    
        font-family: Arial; 
        font-size: 15px;
    
        padding: 5px;
        margin-top: 14px;
    
        resize: none;
    }
    
    #userFeedTextArea:focus{ 
        outline: none;
    }
    
    #sendFeedBtn{ 
        margin-top: 14px;
    
        width: 100px;
        height: 65px;
    
        background: white; 
        
        font-family: Arial; 
        font-size: 18px;
        color: #F97646;
    
        border-style: solid; 
        border-width: 1px;
        border-left-width: 0px; 
        border-color: #F97646;
        border-bottom-right-radius: 10px;
    }
    
    #sendFeedBtn:focus{ 
        outline: none;
    }
    
    #sendFeedBtn:hover{ 
        background: #F97646;
        color: white;
    }
}

@media screen and (max-width: 400px){ 
    #userFeedContainer{ 
        height: 450px;
        width: 300px;
    
        margin-top: 10px;
        margin-left: -15px;

        background: transparent;
    }
    
    #userFeedInnerCont{ 
        height: 410px;
        width: 100%;
    
        background:#f9f9f9;
    
        border-style: none; 
        border-radius: 10px;
    }
    
    #userFeedInnerScroll{ 
        width: 100%;
        height: 360px;
    
        overflow-y: auto;
    
        background:#f9f9f9;
        border-style: none; 
        border-top-left-radius: 10px; 
        border-top-right-radius: 10px;
    }
    
    #userFeedInnerScroll::-webkit-scrollbar{ 
        width: 3px;
    }
    #userFeedInnerScroll::-webkit-scrollbar-thumb{
        background: #F97646;
        border-radius: 2px; 
    }
    
    #userFeedInnerScroll::-webkit-scrollbar{ 
        height: 0px;
    }   
    
    #userFeedInputCont{ 
        width: 100%;
        height: 53px;
    
        background:white;
    
        border-style: none ; 
        border-width: 1px;
        border-color: #F97646;
        border-bottom-left-radius: 10px;
        border-bottom-right-radius: 10px;
    
        display: flex; 
        flex-direction: row; 
        justify-content: left;
        align-items: center;
    }
    
    #userFeedTextArea{ 
        width:300px;
        height: 53px;
    
        border-style: solid; 
        border-bottom-left-radius: 10px;
        border-width: 1px; 
        border-color: #F97646;
    
        font-family: Arial; 
        font-size: 15px;
    
        padding: 5px;
        margin-top: 14px;
    
        resize: none;
    }
    
    #userFeedTextArea:focus{ 
        outline: none;
    }
    
    #sendFeedBtn{ 
        margin-top: 14px;
    
        width: 100px;
        height: 65px;
    
        background: white; 
        
        font-family: Arial; 
        font-size: 18px;
        color: #F97646;
    
        border-style: solid; 
        border-width: 1px;
        border-left-width: 0px; 
        border-color: #F97646;
        border-bottom-right-radius: 10px;
    }
    
    #sendFeedBtn:focus{ 
        outline: none;
    }
    
    #sendFeedBtn:hover{ 
        background: #F97646;
        color: white;
    }
}
