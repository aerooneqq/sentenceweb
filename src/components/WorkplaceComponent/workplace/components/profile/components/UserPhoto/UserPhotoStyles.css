@media screen and (min-width: 400px){ 
  #photoBorder{
    margin-top: 30px;

    width: 200px;
    height: 200px;

    background: white;
  
    border-style: solid;
    border-width: 1px;
    border-radius: 250px;
    border-color: #F97646;

    box-sizing: content-box;

    padding: 0.1px;

    position: relative;
  }

  .userPhotoAndSettingsCont { 
    width: 100%;

    display: flex; 
    flex-direction: column;
    justify-content: center; 
    align-items: center;
  }

    #photoBorder:hover { 
      border-color: orangered;
    }

  .fillContainer { 
    flex-grow: 1;
  }

  .userPhotoTopSettings { 
    width: 100%;
    height: 50px;

    background: #e0e0e0;

    padding-left: 2.5%;
    padding-right: 2.5%;

    box-sizing: border-box;


    display: flex;
    flex-direction: row;  
    justify-content: baseline;
    align-items: center;
  }

  #userPhotoImg { 
    width: 200px;
    height: 200px;

    border-radius: 200px;
  }
  
  #photoContainer{  
    width: 300px; 
    
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
  }
  
  #photoBorder:hover #uploadNewUserPhotoContainer{
    visibility: visible;
    animation: uploadUserPhotoContAnimation 0.5s ease forwards;
  }
  
  @keyframes uploadUserPhotoContAnimation {
    from{
      opacity:  0;
    }
  
    to {
      opacity: 1;
    }
  }
  
  #profileDataItemsContainer{
    display: flex;
    flex-direction: row;
  }
  
  #uploadNewUserPhotoContainer{
    position: absolute;

    top: 0;
    left: 0;
  
    visibility: hidden;
    opacity: 0;
    width: 50px;
    height: 50px;
  
    border-style: none;
    border-radius: 50px;
    background: #DFDFDF;
  
    z-index: 2;
    margin-top: 120px;
    margin-left: 75px
  }
  
  #uploadNewUserPhotoContainer:hover {
    background: #E4E4E4;
  }
  
  #uploadNewUserPhotoImg{
    background: url("../../img/uploadNewUserPhoto.png");
  
    width: 24px;
    height: 24px;

    position: relative;
    left: 13px;
    top: 11px;
  }

  #uploadNewUserPhotoInput { 
    visibility: collapse;
  }
  
  #userPhotoloaderContainer{ 
    width: 100%;
    height: 100%;
  
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
  }
}

@media screen and (max-width: 400px){ 
  #photoBorder{
    width: 200px;
    height: 200px;
  
    background: white;
  
    border-style: solid;
    border-width: 0.7px;
    border-radius: 250px;
    border-color: orangered;
  
    display: flex;
    flex-direction: column;
    align-items: center;
  }

  .userPhotoAndSettingsCont { 
    width: 100%;
    
    display: flex; 
    flex-direction: column;
    justify-content: center; 
    align-items: center;
  }

  .fillContainer { 
    flex-grow: 1;
  }

  .userPhotoTopSettings { 
    width: 100%;
    height: 50px;

    padding-left: 2.5%;
    padding-right: 2.5%;

    box-sizing: border-box;

    background: #e0e0e0;

    display: flex;
    flex-direction: row;  
    justify-content: baseline;
    align-items: center;
  }
  
  #photoContainer{ 
    margin-top: 20px;

    display: flex;
    flex-direction: column;
    justify-content: center; 
    align-items: center;
  }
  
  #photoBorder:hover #uploadNewUserPhotoContainer{
    visibility: visible;
    animation: uploadUserPhotoContAnimation 0.5s ease forwards;
  }
  
  @keyframes uploadUserPhotoContAnimation {
    from{
      opacity:  0;
    }
  
    to {
      opacity: 1;
    }
  }
  
  #profileDataItemsContainer{
    display: flex;
    flex-direction: row;
  }
  
  #uploadNewUserPhotoContainer{
    position: relative;
  
    visibility: hidden;
    opacity: 0;
    width: 50px;
    height: 50px;
  
    border-style: none;
    border-radius: 50px;
    background: #DFDFDF;
  
    z-index: 2;
    margin-top: 140px;
  }
  
  #uploadNewUserPhotoContainer:hover {
    background: #E4E4E4;
  }
  
  #uploadNewUserPhotoImg{
    background: url("../../img/uploadNewUserPhoto.png");
  
    width: 24px;
    height: 24px;
  
    position: relative;
    left: 13px;
    top: 11px;
  }

  #uploadNewUserPhotoInput { 
    visibility: collapse;
  }
  
  #loaderContainer{ 
    width: 100%;
    height: 100%;
  
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
  }
}