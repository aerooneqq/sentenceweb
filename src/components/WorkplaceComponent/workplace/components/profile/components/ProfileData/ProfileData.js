import React, {Component, lazy, Suspense} from "react";

//Styles
import "./ProfileDataContStyles.css";

//Model
import ProfileDataModel from "./ProfileDataModel";


//Components
import ProfileDataLoader from "./ProfileDataLoader/ProfileDataLoader";
import ProfileDataHeader from "./profileDataAtoms/ProfileDataHeader/ProfileDataHeader";
import ProfileHeader from "../ProfileHeader/ProfileHeader";
import Loader from "../../../../../../loader/Loader";

const profileDataModel = new ProfileDataModel();

/**
 * The brief description of that class and components' dependencies:
 * 1) All profile text blocks are wrapped in a "textBlock" class, the style rules for this
 *    class are written in the ProfileDataContStyles.css
 * 2) All logic is placed in the ProfileDataModel class.
 * 
 * */
export default class ProfileData extends Component {
  constructor(props) {
    super(props);

    this.state = {
      isFirstLoad: true,
      isUpdating: true,
      userFriendsMode: "subscribers"
    };

    this.changeUpdatedUser = this.changeUpdatedUser.bind(this);
    this.discardChangesInProfileData = this.discardChangesInProfileData.bind(this);
    this.setUpdatingStatus = this.setUpdatingStatus.bind(this);
    this.changeUserFriendMode = this.changeUserFriendMode.bind(this);
    
    profileDataModel.setUpdateUserFunction(this.changeUpdatedUser).configureModel();
  }

  setUpdatingStatus(status) { 
    this.setState({ 
      isUpdating: status
    });
  }

  changeUserFriendMode(mode){ 
    this.setState({ 
      userFriendsMode: mode
    })
  }

  discardChangesInProfileData() { 
    profileDataModel.discardAllChanges();
  }

  changeUpdatedUser(propertyName, value) {
    profileDataModel.changeUpdatedUser(propertyName, value);
  }

  render() {
    if (this.state.isFirstLoad === true && this.props.user === null) { 
      return (
        <div id = "profileDataOutterContainer">
          <ProfileHeader header = "Settings"/>
          <div id = "profileDataContainer">
            <ProfileDataLoader />
          </div>  
        </div>
      )
    }
    else { 
      return (
          <div id = "profileDataOutterContainer">
            <ProfileHeader header = "Settings"/>
            <div id = "profileDataContainer">
                <div id = "profileDataHeaderCont">
                  <ProfileDataHeader dataName = {profileDataModel.getProfileDataName(this.props.currentUserDataIndex)}/>
                 </div>  
          
                <div className = "profileDataContentCont">
                    {profileDataModel.getProfileDataContent(this.props.currentUserDataIndex)}
                </div>
            </div>
          </div>
      )
    }
  }
}
