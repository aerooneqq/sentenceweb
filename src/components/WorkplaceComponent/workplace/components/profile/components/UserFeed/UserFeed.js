import React, {Component, lazy, Suspense} from "react"

//Styles
import "./UserFeedStyles.css"

//Services
import UserFeedService from "../../../../../../../services/UserServices/UserFeedService";
import ProfileHeader from "../ProfileHeader/ProfileHeader";

//App messages
import {alertAppMessage} from "../../../../../../ApplicationMessage/ApplicationMessageManager";

//Components
const UserAtomFeed = lazy(()=>import("./UserAtomFeed/UserAtomFeed"));
const Loader = lazy(() => import("../../../../../../loader/Loader"));

export default class UserFeed extends Component { 

    constructor(props) { 
        super(props)

        this.state = { 
            component: <Loader message = "Loading feed..." />,
            textAreaValue: ""
        };

        this.userFeedService = new UserFeedService(localStorage.getItem("token"));

        this.insertNewPost = this.insertNewPost.bind(this);
        this.onTextAreaValueChange = this.onTextAreaValueChange.bind(this);
        this.uploadUserFeed = this.uploadUserFeed.bind(this);
        this.onTextAreaKeyDown = this.onTextAreaKeyDown.bind(this);

        this.uploadUserFeed();
    }

    /**
     * Deletes all repeated entries of elements
     * @param {Array of numbers} array 
     */
    _clearArray(array) { 
        var seen = {};
        var out = [];
        var len = array.length;
        var j = 0;

        for(var i = 0; i < len; i++) {
             var item = array[i];
             if(seen[item] !== 1) {
                   seen[item] = 1;
                   out[j++] = item;
             }
        }

        return out
    }

    uploadUserFeed() { 
        this.userFeedService.getUserFeed(localStorage.getItem("token"))
            .then(res => { 
                let atomFeeds = res.data.map(f => {
                    return  (
                        <UserAtomFeed date = {f.publicationDate}
                                      message = {f.message}
                                      photo = {f.photo}
                                      authorName = {f.name + " " + f.surname}/>
                    );
                });

                atomFeeds.reverse();

                this.setState({
                    component: atomFeeds,
                    textAreaValue: ""
                }); 
            }).catch(er => {
                if (er.response) { 
                    alertAppMessage(er.reponse.data, "error");
                }
                else { 
                    alertAppMessage("Error occured while getting your feed", "error");
                }
            });
    }

    insertNewPost() { 
        this.setState({ 
            component: <Loader message = "Inserting your message" />
        });

        let message = this.state.textAreaValue;

        this.userFeedService.insertUserPost(message).then(() => { 
                alertAppMessage("The message was posted!", "success");
                this.uploadUserFeed();
            }).catch(er => { 
                if (er.response) { 
                    alertAppMessage(er.reponse.data, "error");
                }
                else { 
                    alertAppMessage("The error occured while inserting your message", "error");
                }
            });

    }

    onTextAreaValueChange(event) { 
        this.setState({ 
            textAreaValue: event.target.value
        });
    }

    onTextAreaKeyDown(event) { 
        if (event.keyCode === 13) { 
            this.insertNewPost();
        }
    }

    render() { 
        return(
            <div id = "userFeedContainer">
                <ProfileHeader header = "Feed" /> 
                <div id = "userFeedInnerCont">
                    <Suspense fallback={<Loader />}>
                        <div id = "userFeedInnerScroll">
                            {this.state.component}
                        </div>
                        <div id = "userFeedInputCont">
                            <textarea placeholder="Whats up? Type it here..."
                                      id = "userFeedTextArea"
                                      value = {this.state.textAreaValue}
                                      onChange = {this.onTextAreaValueChange}
                                      onKeyDown = {this.onTextAreaKeyDown}/>
                            <button id = "sendFeedBtn" onClick = {this.insertNewPost}>
                                Send
                            </button>
                        </div>
                    </Suspense>
                </div>
            </div>
        )
    }
}