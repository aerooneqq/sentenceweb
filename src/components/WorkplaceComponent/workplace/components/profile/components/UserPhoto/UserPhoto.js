import React, {Component} from "react";

//Styles
import "./UserPhotoStyles.css";

//Icons
import careerIcon from "./img/career_profile_data_item.svg";
import authorizationIcon from "./img/authorization_profile_data_item.svg";
import friendsIcon from "./img/friends_profile_data_item.svg"; 
import locationIcon from "./img/location_profile_data_item.svg";
import nameIcon from "./img/name_profile_data_item.svg";

//Services
import { alertAppMessage } from "../../../../../../ApplicationMessage/ApplicationMessageManager";
import ResponseService from "../../../../../../../services/ResponseService/ReponseService";
import UserPhotoService from "../../../../../../../services/UserPhotoService/UserPhotoService";

//Components
import Loader from "../../../../../../loader/Loader";
import UserPhotoLoader from "./UserPhotoLoader/UserPhotoLoader";
import ProfileDataItem from "./ProfileDataItem/ProfileDataItem";
import SignOutComponent from "./SignOut/SignOut";
import DeleteAccountComponent from "./DeleteAccount/DeleteAccount";
import AccountVerification from "./AccountVerification/AccountVerification";

/**
 * PROPS LIST:
 * 1) changeUserData - the function to change the currently displayed data.
 * 2) signOut  - the function to sign out from the system
 */
export default class UserPhoto extends Component {
  _icons = [careerIcon, authorizationIcon, nameIcon, locationIcon, friendsIcon];
  _toolTips = ["Career", "Authentication", "Name", "Location", "Subscribers and subscriptions"];

  constructor(props) {
    super(props);

    this.userPhotoService = new UserPhotoService(localStorage.getItem("token"));
    this.responseService = new ResponseService();

    let profileDataItems = [];

    for (let i = 0; i < 5; i++){ 
      let props = { 
        imgURL: careerIcon,
        currentUserData: i,
        changeUserData: this.props.changeUserData,

      }

      profileDataItems.push(<ProfileDataItem data = {props} 
                                             icon = {this._icons[i]}
                                             toolTip = {this._toolTips[i]}/>);
    }

    this.state = { 
      profileDataItems: profileDataItems,
      isPhotoUpdating: true,
      isFirstLoad: true,
      userPhoto: null
    }

    this.updateUserPhoto = this.updateUserPhoto.bind(this);
    this.uploadUserPhoto = this.uploadUserPhoto.bind(this);
  }

  componentDidMount() { 
    this.uploadUserPhoto();
  }

  uploadUserPhoto() { 
    this.userPhotoService.getUserPhoto().then(res => { 
      this.setState({ 
        isPhotoUpdating: false,
        isFirstLoad: false,
        userPhoto: res.data.photo
      });

      document.getElementById("userPhotoImg").src = `data:image/png;base64, ${this.state.userPhoto}`
    }).catch(er => { 
      this.setState({ 
        isPhotoUpdating: false,
        isFirstLoad: false,
      });
      this.responseService.alertErrorMessage(er, "Error occured while getting photo.");
    });
  }

  updateUserPhoto() { 
    let input = document.createElement("input");
    input.type = "file";
    input.accept = "image/jpeg,image/png,image/gif";

    input.onchange = e => { 
      let file = e.target.files[0];

      let reader = new FileReader();
      reader.readAsArrayBuffer(file);

      reader.onload = e => { 
        let array = new Uint8Array(e.target.result);
        
        let byteArray = []

        for (let i = 0; i < array.length; i++) { 
          byteArray.push(array[i]);
        }

        this.setState({ 
          isPhotoUpdating: true
        });

        this.userPhotoService.updateUserPhoto(byteArray).then(() => { 
          alertAppMessage("The profile photo was updated!", "success");
          this.setState({ 
            isPhotoUpdating: false
          });
        }).catch(er => { 
          this.responseService.alertErrorMessage(er, "The error occured when updating the photo");
          this.setState({ 
            isPhotoUpdating: false
          });
        });
      }
    }

    input.click();
  }

  render() {
      return (
        <div id="photoContainer">
          
          {this.state.isFirstLoad === true ? <UserPhotoLoader /> :
              <div className = "userPhotoAndSettingsCont">
                <div className = "userPhotoTopSettings">
                  <SignOutComponent signOut = {this.props.signOut}/>
                  <div className = "fillContainer" />
                  <DeleteAccountComponent />
                </div>
      
                <div id="photoBorder">
                  {this.state.isPhotoUpdating === true ? <Loader message = "Updating the photo..."/> : 
                  <div>
                    <img id = "userPhotoImg" alt = "Profile" />
                    <div id ="uploadNewUserPhotoContainer" onClick = {this.updateUserPhoto}>
                      <div id="uploadNewUserPhotoImg"></div>
                    </div> 
                  </div>}
      
                </div>
                <div id="profileDataItemsContainer">
                  {this.state.profileDataItems}
                </div>
              </div>
          }
          <AccountVerification />
        </div>
      )
  }
}
