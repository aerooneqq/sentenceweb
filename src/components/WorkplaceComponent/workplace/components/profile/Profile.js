import React, {Component, Suspense, lazy} from "react"
import "./ProfileStyles.css"

//Services
import UserService from "../../../../../services/userServices/UserService"
import UserActivityService from "../../../../../services/userServices/UserActivitiesService"
import ProfileData from "./components/ProfileData/ProfileData";

//Components
const UserPhoto = lazy(()=>import("./components/UserPhoto/UserPhoto"))
const UserActivityComponent = lazy(()=>import("./components/UserActivity/UserActivity"))
const UserFeed = lazy(()=>import("./components/UserFeed/UserFeed"))

export default class Profile extends Component{
  
  constructor(props){
    super(props);

    this.state = { 
      currentUserDataIndex: 2
    };

    this.userService = new UserService();

    this.changeCurrentUserData = this.changeCurrentUserData.bind(this);
  }

  changeCurrentUserData(newUserDataIndex){
    this.setState({ 
      currentUserDataIndex: newUserDataIndex
    });
  }

  render(){
    return (
      <div id = "profileContainer">
        <div id = "userPhotoCont">
            <UserPhoto changeUserData = {this.changeCurrentUserData} />
        </div>

        <div id = "rightProfileCont">
          <div id = "userActivitiesOutterCont">
            <UserActivityComponent />
          </div>

          <div id = "userSettingsAndFeedOutterCont">
            <ProfileData currentUserDataIndex = {this.state.currentUserDataIndex} />
            <UserFeed />
          </div>
        </div>
      </div>
    )
  }
}
