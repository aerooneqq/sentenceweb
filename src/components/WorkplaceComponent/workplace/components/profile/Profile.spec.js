import React from 'react'
import { shallow, render } from 'enzyme'
import {shallowToJson} from "enzyme-to-json"
import Profile from "./Profile";
import ProfileData from "./components/ProfileData/ProfileData";
import UserPhoto from "./components/UserPhoto/UserPhoto";
import UserActivityComponent from "./components/UserActivity/UserActivity";
import UserFeed from "./components/UserFeed/UserFeed";

describe("Profile", () => { 
    it("should render correctly", () => { 
        const props = { 
            signOut: () => {}
        }

        const profile = shallow(<Profile {...props}/>);

        expect(shallowToJson(profile)).toMatchSnapshot();
    });

    it("should change the currentUserDataIndex", () => { 
        const profile = shallow(<Profile />);

        expect(profile.state("currentUserDataIndex")).toBe(2);
        
        profile.instance().changeCurrentUserData(4);

        expect(profile.state("currentUserDataIndex")).toBe(4);
    });

    it("should have profile mini components", () => { 
        const profile = shallow(<Profile />);

        expect(profile.find(UserPhoto).exists()).toBe(true);
        expect(profile.find(ProfileData).exists()).toBe(true);
        expect(profile.find(UserActivityComponent).exists()).toBe(true);
        expect(profile.find(UserFeed).exists()).toBe(true);
    }); 
});