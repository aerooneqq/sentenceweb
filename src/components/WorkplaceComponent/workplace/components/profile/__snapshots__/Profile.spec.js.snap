// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile should render correctly 1`] = `
<div
  id="profileContainer"
>
  <div
    id="userPhotoCont"
  >
    <UserPhoto
      changeUserData={[Function]}
      signOut={[Function]}
    />
  </div>
  <div
    id="rightProfileCont"
  >
    <div
      id="userActivitiesOutterCont"
    >
      <UserActivity />
    </div>
    <div
      id="userSettingsAndFeedOutterCont"
    >
      <ProfileData
        currentUserDataIndex={2}
      />
      <UserFeed />
    </div>
  </div>
</div>
`;
